cli/crates/cli/tests/utils/environment.rs

```diff
@@ -1,23 +1,25 @@
 #![allow(dead_code)]
 
+use super::async_client::AsyncClient;
 use super::kill_with_children::kill_with_children;
 use super::{cargo_bin::cargo_bin, client::Client};
 use duct::{cmd, Handle};
+use std::sync::Arc;
 use std::{env, fs, io::Write, path::PathBuf};
 use tempfile::{tempdir, TempDir};
 
 pub struct Environment {
     pub endpoint: String,
     directory: PathBuf,
-    temp_dir: TempDir,
+    temp_dir: Arc<TempDir>,
```

Comment:
<img width="592" alt="Screen Shot 2022-08-16 at 12 19 59" src="https://user-images.githubusercontent.com/14347895/184856570-9af84b51-85ab-4ec7-976a-8b085d749d04.png">
(should be fine in terms of performance as it's a test struct)